web/components/contract-search.tsx

```diff
@@ -80,6 +92,105 @@ export function ContractSearch(props: {
     headerClassName,
   } = props
 
+  const [numPages, setNumPages] = useState(1)
+  const [pages, setPages] = useState<Contract[][]>([])
+  const [showTime, setShowTime] = useState<ShowTime | undefined>()
+
+  const searchParameters = useRef<SearchParameters | undefined>()
+  const requestId = useRef(0)
+
+  const performQuery = async (freshQuery?: boolean) => {
```

Comment:
Wouldn't it be nice to make this take an argument of `searchParameters`, which could include the boolean `freshQuery` as a prop?

Then you wouldn't need the Ref.

Reply:
I believe I would need the ref, because we need to know "the last search parameters we used to load the grid" for when you scroll to the bottom and want to load a new page. Do you have a different plan?

Reply:
Oh, I see. I don't have another plan!